rviz_default_plugins/test/rviz_default_plugins/publishers/image_publisher.hpp

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2018, Bosch Software Innovations GmbH.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the Willow Garage, Inc. nor the names of its
+ *       contributors may be used to endorse or promote products derived from
+ *       this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef RVIZ_DEFAULT_PLUGINS__PUBLISHERS__IMAGE_PUBLISHER_HPP_
+#define RVIZ_DEFAULT_PLUGINS__PUBLISHERS__IMAGE_PUBLISHER_HPP_
+
+#include <cmath>
+#include <iostream>
+#include <vector>
+
+#include "opencv2/core/core.hpp"
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp/clock.hpp"
+#include "sensor_msgs/msg/image.hpp"
+#include "std_msgs/msg/header.hpp"
+
+using namespace std::chrono_literals;  // NOLINT
+
+namespace nodes
+{
+
+class ImagePublisher : public rclcpp::Node
+{
+public:
+  ImagePublisher()
+  : Node("path_publisher")
+  {
+    publisher = this->create_publisher<sensor_msgs::msg::Image>("image");
+    timer = this->create_wall_timer(100ms, std::bind(&ImagePublisher::timer_callback, this));
+  }
+
+private:
+  void timer_callback()
+  {
+    auto message = sensor_msgs::msg::Image();
+    message.header = std_msgs::msg::Header();
+    message.header.frame_id = "image_frame";
+    message.header.stamp = rclcpp::Clock().now();
+
+    cv::Mat image(200, 300, CV_8UC4, cv::Scalar(0, 200, 0));
```

Comment:
Is this the first place OpenCV is being used? It appears to be the only place so far. If this is the only place you see it being used for now it would be great to avoid using OpenCV for this. It appears to only be creating a green image right?

Reply:
I should say, it would be great to avoid OpenCV even as just a test dependency since it is quite a large dependency. We've tried pretty hard to avoid a dependency in rviz so far. However, if you foresee needing it in more serious circumstances in the future then a test dependency is not the end of the world.

Reply:
Fine with me, we removed the dependency.